converter.js

```diff
@@ -405,9 +405,11 @@ class Converter {
         // Prove otherwise? either a normal returns or as an async promise
         if (func.returns) {
             returnType = this.convertType(func.returns);
+            if (func.returns.optional && (returnType !== 'any' || returnType !== 'void')) returnType += ' | void';
```

Comment:
i'd do it like this:

```typescript
let alreadyOptional = ['any', 'undefined', 'void'];
if (func.returns.optional && !alreadyOptional.includes(returnType)) returnType += ' | void';
```

the condition in yours isn't very clear and i'm not sure if it works

Reply:
It works as far as I can tell, but I do like your method better (though I'd perhaps the alreadyOptional as a constant since it's used in several places)